cg/cli/workflow/taxprofiler/base.py

```diff
@@ -28,19 +38,89 @@ def taxprofiler(context: click.Context) -> None:
     )
 
 
+taxprofiler.add_command(resolve_compression)
+
+
 @taxprofiler.command("config-case")
 @ARGUMENT_CASE_ID
 @OPTION_INSTRUMENT_PLATFORM
+@DRY_RUN
 @click.pass_obj
-def config_case(context: CGConfig, case_id: str, instrument_platform: SequencingPlatform) -> None:
-    """Create sample sheet file for Taxprofiler analysis for a given case_id."""
+def config_case(
+    context: CGConfig, case_id: str, instrument_platform: SequencingPlatform, dry_run: bool
+) -> None:
+    """Create sample sheet file for Taxprofiler analysis for a given case."""
     analysis_api: TaxprofilerAnalysisAPI = context.meta_apis[MetaApis.ANALYSIS_API]
 
     try:
         analysis_api.status_db.verify_case_exists(case_internal_id=case_id)
-        analysis_api.config_case(case_id=case_id, instrument_platform=instrument_platform, fasta="")
+        analysis_api.config_case(
+            case_id=case_id, instrument_platform=instrument_platform, dry_run=dry_run
+        )
 
     except CgError as error:
-        LOG.error(f"Could not create sample sheet: {error}")
         LOG.error(f"{case_id} could not be found in StatusDB!")
         raise click.Abort() from error
+
+
+@taxprofiler.command("run")
+@ARGUMENT_CASE_ID
+@OPTION_LOG
+@OPTION_WORKDIR
+@OPTION_FROM_START
+@OPTION_PROFILE
+@OPTION_CONFIG
+@OPTION_PARAMS_FILE
+@OPTION_REVISION
+@OPTION_USE_NEXTFLOW
+@DRY_RUN
+@click.pass_obj
+def run(
+    context: CGConfig,
+    case_id: str,
+    log: str,
+    work_dir: str,
+    from_start: bool,
+    profile: str,
+    config: str,
+    params_file: str,
+    revision: str,
+    use_nextflow: bool,
+    dry_run: bool,
+) -> None:
+    """Run taxprofiler analysis for given CASE ID."""
+    analysis_api: TaxprofilerAnalysisAPI = context.meta_apis[MetaApis.ANALYSIS_API]
+    analysis_api.status_db.verify_case_exists(case_internal_id=case_id)
+
+    command_args: CommandArgs = CommandArgs(
```

Comment:
In which folder could this utility function be added?

Reply:
A suggestion is `cg/cli/workflow/taxprofiler/utils/taxprofler_command.py`

Reply:
Since `CommandArgs` is a class within `rnafusion` and it is general, can it maybe be under `cg/cli/workflow/utils` instead to be shared between taxprofiler and rnafusion in an upcoming PR?

Reply:
I will merge and deploy now if you are satisfied with the rest of the changes :)